src/context.rs

```diff
@@ -741,10 +1033,69 @@ impl DefaultSeafowlContext {
             .ok_or_else(|| {
                 Error::Plan(format!("Schema {schema_name:?} does not exist!"))
             })?;
-        Ok(self
-            .table_catalog
-            .create_table(collection_id, &table_name, &sf_schema)
-            .await?)
+
+        let delta_schema = DeltaSchema::try_from(schema)?;
+
+        // TODO: we could be doing this inside the DB itself (i.e. `... DEFAULT gen_random_uuid()`
+        // in Postgres and `... DEFAULT (uuid())` in SQLite) however we won't be able to do it until
+        // sqlx 0.7 is released (which has libsqlite3-sys > 0.25, with the SQLite version that has
+        // the `uuid()` function).
+        // Then we could create the table in our catalog first and try to create the delta table itself
+        // with the returned uuid (and delete the catalog entry if the object store creation fails).
+        // On the other hand that would complicate etag testing logic.
+        #[cfg(test)]
+        let table_uuid = deterministic_uuid();
```

Comment:
This is kind of ugh to have this test code live here. Is there some way we can inject this into the Context struct when it's created?

Reply:
The alternatives that I see are:
1. Hide this behind a common indirection, e.g. just call `get_uuid()`, and then it would ultimately do the same thing as above.
2. Extend `DefaultSeafowlContext` with another field.

I feel that `DefaultSeafowlContext` struct is crammed enough already, so I dislike option #2 more than the current approach. Also, if we're bound to change the etag calculation logic this may not be necessary very soon, thus I'm inclined to keep it as is.

Reply:
Makes sense!